testhelpers/docker_registry.go

```diff
@@ -88,6 +96,27 @@ func ReadOnly(handler http.Handler) http.Handler {
 	})
 }
 
+func Custom(handler http.Handler, permissions map[string]*ImagePrivileges) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if permission, ok := permissions[extractImageName(r.URL.Path)]; ok {
+			if r.Method == "GET" || r.Method == "HEAD" {
+				if !permission.readable {
```

Comment:
I would add a comment in each of these conditions to make the flow clearer. Something like:
```
// read request was arrived while read permission isn't allowed
```

Reply:
Done. [commit](https://github.com/buildpacks/imgutil/pull/125/commits/a0fd89a42b3519eaf6dc613931da5b34d11cf397)